rasa/server.py

```diff
@@ -227,27 +227,85 @@ def event_verbosity_parameter(
         )
 
 
-async def get_tracker(
+async def get_tracker_with_session_start(
     processor: "MessageProcessor", conversation_id: Text
 ) -> DialogueStateTracker:
-    """Get tracker object from `MessageProcessor`."""
-    tracker = await processor.get_tracker_with_session_start(conversation_id)
-    _validate_tracker(tracker, conversation_id)
+    """Get tracker object from `MessageProcessor` and update the conversation session.
 
-    # `_validate_tracker` ensures we can't return `None` so `Optional` is not needed
-    return tracker  # pytype: disable=bad-return-type
+    Args:
+        processor: An instance of `MessageProcessor`.
+        conversation_id: Conversation ID to fetch the tracker for.
 
+    Returns:
+        The tracker for `conversation_id` with an updated conversation session.
+    """
+    return await processor.get_tracker_with_session_start(conversation_id)
 
-def _validate_tracker(
-    tracker: Optional[DialogueStateTracker], conversation_id: Text
-) -> None:
-    if not tracker:
-        raise ErrorResponse(
-            409,
-            "Conflict",
-            f"Could not retrieve tracker with ID '{conversation_id}'. Most likely "
-            f"because there is no domain set on the agent.",
-        )
+
+def get_test_stories(
+    processor: "MessageProcessor",
+    conversation_id: Text,
+    until_time: Optional[float],
+    fetch_all_sessions: bool = False,
+) -> Text:
+    """Retrieves test stories from `processor` for all conversation sessions for
+    `conversation_id`.
+
+    Args:
+        processor: An instance of `MessageProcessor`.
+        conversation_id: Conversation ID to fetch stories for.
+        until_time: Timestamp up to which to include events.
+        fetch_all_sessions: Whether to fetch stories for all conversation sessions.
+            If `False`, only the last conversation session is retrieved.
+
+    Returns:
+        The stories for `conversation_id` in test format.
+    """
+    if fetch_all_sessions:
+        trackers: List[
+            DialogueStateTracker
+        ] = processor.get_trackers_for_all_conversation_sessions(conversation_id)
```

Comment:
shouldn't we raise in case the `conversation_id` is invalid?

Reply:
yeah we should but this goes for all endpoints that depend on the `conversation_id`. I would suggest handling this separately, for example in a decorator that uses the processor to validate that `conversation_id` exists

Reply:
no hang on, the function was just badly named before: 

https://github.com/RasaHQ/rasa/pull/6979/files#diff-394c9a945911d29e59b1eae237cdf29d7bc1276940b6dfc2cd95dd5a51117143L230

it was still executing the session start

Reply:
Sounds good. Can you please create an issue for that?

Reply:
https://github.com/RasaHQ/rasa/issues/7022